File: year2020/month12/day1223/ms_1223.cpp
#include <iostream>

using namespace std;

bool solution(int x);

int main() {
	int x;

	cin >> x;

	if (1 <= x && x <= 10000) {
		cout << solution(x) << endl;
	}
}

bool solution(int x) {
	int SED[5] = { 0, 0, 0, 0, 0 };
	int temp = x;
	int sum = 0;
	int i = 0;

	while (temp > 0) {
		SED[i] = temp % 10;
		temp /= 10;
		i++;
	}

	for (i = 0; i < 5; i++) {
		sum += SED[i];
	}

	if (!(x % sum))
		return true;
	else
		return false;
}

File: year2020/month12/day1223/ms_1223.cs
using System;

namespace algorithm_128
{
    class ms_1223
    {
        static bool solution(int x)
        {
            int[] SED = new int[5] { 0, 0, 0, 0, 0 }; // Sum of Each Digit
            int sum = 0;
            int i = 0;
            int temp = x;

            while(temp > 0)
            {
                SED[i] = temp % 10;
                temp /= 10;
                i++;
            }

            foreach(var data in SED)
            {
                sum += data;
            }

            if(x % sum == 0)
            {
                return true;
            }
            else
            {
                return false;
            }  
        }

        static void Main(string[] args)
        {
            int x = Convert.ToInt32(Console.ReadLine());

            if(1 <= x && x <= 10000)
            {
                Console.WriteLine(solution(x));
            }
        }
    }
}
